### PR TITLE
Enchance documentation

### PR DESCRIPTION
**Pull request by SunSince90**
This PR enhancing documentation by splitting it into multiple files and making it much more organized.

Additionally, this also updates version format to `0.x.x` and adds badges to the repository.

### CODE REVIEW

File: docs/usage.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+In the following example, the CNWAN Reader watches changes in
+Google Cloud Service Directory with the following requirements:
+
+* The *allowed* services have at least the `key-name` key in their metadata
+* The project is called `my-project`
+* The region is `us-west2`
+* Service account is placed inside `path/to/creds` folder
+* The name of the service account file is `serv-acc.json`
+* The endpoint of the adaptor is `http://example.com:5588/my/path`
```
Comment (arnatal) [inline docs/usage.md:91]: I'd use the default as example here i.e. "localhost:80"

File: docs/usage.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+example that includes a CNWAN Adaptor endpoint as well.
+
+### Metadata Key
+
+The CNWAN Reader only reads services that have the provided metadata key.
+
+For example, the following flag
+
+```bash
+--metadata-key prefix/key
```
Comment (arnatal) [inline docs/usage.md:53]: Maybe we should provide an example related to the general CNWAN project, like "cwan.io/traffic-profile", so people coming to the Reader from one of our demos/presos can easily map what this is for.

**Review by arnatal — CHANGES_REQUESTED** [review]
This looks very good Elis, thanks for the changes! I've added some minor comments, but honestly I think we can probably merge the PR as is :)

File: README.md [hunk 3]
```diff
@@ -12,18 +19,7 @@ You can contact the CNWAN team at [cnwan@cisco.com](mailto:cnwan@cisco.com).
```
[the remark this answers is not in the document]
Comment (SunSince90) [reply README.md:19]: Yeah I think that we can remove it, will be much cleaner like this

File: docs/docker_installation.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  [Docker Desktop for Mac](https://hub.docker.com/editions/community/docker-ce-desktop-mac/)
+* *Windows* users:
+  [Docker Desktop for Windows](https://hub.docker.com/editions/community/docker-ce-desktop-windows/)
+
+## Use the official image
+
+Run
+
+```bash
+docker pull cnwan/cnwan-reader:v0.3.0
```
[the remark this answers is not in the document]
Comment (SunSince90) [reply docs/docker_installation.md:28]: Yes, #12 introduces a workflow that auto-builds&pushes a docker image whenever a new tag/release is created :)

File: docs/docker_installation.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+Follow the [Docker Usage](./docker-usage.md) section to learn how to use it.
+
+## Optional: Rename it
+
+If you want to avoid specifying its full repository name, you can rename it.
+
+Execute:
+
+```bash
+docker rename cnwan/cnwan-reader cnwan-reader
```
[the remark this answers is not in the document]
Comment (SunSince90) [reply docs/docker_installation.md:101]: This section is unrelated from the previous and now that I read it I realize that it isnt very straightforward. I will modify it accordingly. Thank you!

File: docs/docker_usage.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+```bash
+docker run \
+-v ~/Desktop/cnwan-credentials/serv-acc.json:/credentials/serv-acc.json \
+my-image \
+servicedirectory \
+--project my-project \
+--region us-west2 \
+--metadata-key cnwan.io/traffic-profile \
+--interval 10 \
+--service-account ./credentials/serv-acc.json
```
Comment (arnatal) [inline docs/docker_usage.md:61]: Let's add the --adaptor-api even if we are using the default and it's technically not needed, for clarity
```suggestion
--service-account ./credentials/serv-acc.json
--adaptor-api localhost:8080/cnwan
```

File: docs/usage.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+Google Cloud Service Directory with the following requirements:
+
+* The *allowed* services have at least the `cnwan.io/traffic-profile`
+key in their metadata
+* The project is called `my-project`
+* The region is `us-west2`
+* Service account is placed inside `path/to/creds` folder
+* The name of the service account file is `serv-acc.json`
+* The endpoint of the adaptor is the default one
+(`http://localhost/cnwan/events`), so there is no need to use `--adaptor-api`
```
Comment (arnatal) [inline docs/usage.md:100]: I'd add the --adaptor-api anyways to the command below. Maybe making the comment here that it's not needed when we use the default value but we add it for clarity.

**Review by arnatal — CHANGES_REQUESTED** [review]
Two nano fixes and we should be good to go 😄

**Comment by arnatal** [comment]
I just realized that we should also update the version printed via the 'version' command.

**Comment by SunSince90** [comment]
> I just realized that we should also update the version printed via the 'version' command.

Good catch! 

Also, please join #12 before this one :)

**Pull request merged**